### PR TITLE
Dzelge fix carray drop issue

### PR DESCRIPTION
**Pull request by dzelge**
When accepting this PR:
- xarray 0.14.1 will be the minimum version for xcube's pythin environment
- all usages of xarray.dataset.drop methods will be replaced by xarray.dataset.drop_vars
- a couple of typing issues due to changes in version 0.14.1 will be fixed

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -17,6 +17,8 @@
   hidden `xcube apply` command. (#167) 
 * Added new function `xcube.core.geom.rasterize_features()` 
   to rasterize vector-data features into a dataset. (#222)
+* Made xarray version 0.14.1 minimum requirement due to dropping xarray's drop
```
Comment (forman) [inline CHANGES.md]:
```suggestion
* Made xarray version 0.14.1 minimum requirement due to deprecation of xarray's `Dataset.drop`
```

File: CHANGES.md [hunk 2]
```diff
@@ -17,6 +17,8 @@
   hidden `xcube apply` command. (#167) 
 * Added new function `xcube.core.geom.rasterize_features()` 
   to rasterize vector-data features into a dataset. (#222)
+* Made xarray version 0.14.1 minimum requirement due to dropping xarray's drop 
+  method and replacing it with drop_sel and drop_vars.
```
Comment (forman) [inline CHANGES.md]:
```suggestion
  method and replaced it with `drop_sel` and `drop_vars` accordingly. 
```

File: environment.yml [hunk 3]
```diff
@@ -1,38 +1,29 @@
-name: xcube
+name: xcube5
```
Comment (forman) [inline environment.yml]: ???

File: environment.yml [hunk 4]
```diff
@@ -1,38 +1,29 @@
...
+  - matplotlib >=3.0.3
   - netcdf4 >=1.5.0
   - numba >=0.43
-  - numpy >=1.16
+  # - numpy >=1.16
   - pandas >=0.24
   - pillow >=6.0
-  - proj4 >=5.2
-  - pyproj >=2.1
+  # - proj4 >=5.2
```
Comment (forman) [inline environment.yml]: Please comment, what you did here. Why are `proj4 ` and `pyproj` no longer in?

File: environment.yml [hunk 5]
```diff
@@ -1,38 +1,29 @@
...
+  - python
   - affine >=2.2
-  - blas =*=mkl
   - click >=7.0
   - cmocean >=2.0
   - dask >=1.2
   - fiona >=1.8
-  - gdal >=2.4,<3.0
-  - matplotlib >=3.0
+  # - gdal >=2.4
```
Comment (forman) [inline environment.yml]: Why?

File: environment.yml [hunk 6]
```diff
@@ -1,38 +1,29 @@
-name: xcube
+name: xcube5
 channels:
   - conda-forge
-  - defaults
 dependencies:
   # Python
-  - python >=3.6
-  # Required
+  - python
```
Comment (forman) [inline environment.yml]: Why remove `# Required` comment, why don't we need a minimum Python?

File: environment.yml [hunk 7]
```diff
@@ -1,38 +1,29 @@
...
-  - xarray >=0.12.3
+  - xarray >=0.14.1
   - zarr >=2.3.2
-  - sphinx >=2.0.1
-  # Testing
-  - flake8 >=3.7
-  - pytest >=4.4
-  - pytest-cov >=2.6
-
+  - pytest
```
Comment (forman) [inline environment.yml]: Why no `flake8`, why no longer `pytest-cov`?

File: environment.yml (hunk 4 above)
Comment (dzelge) [reply environment.yml]: Only change now is xarray version

File: environment.yml (hunk 5 above)
Comment (dzelge) [reply environment.yml]: Only chane now is xarray version

File: environment.yml (hunk 6 above)
Comment (dzelge) [reply environment.yml]: Only change now is xarray version

File: environment.yml [hunk 8]
```diff
@@ -28,11 +28,10 @@ dependencies:
...
-  - xarray >=0.12.3
+  - xarray >=0.14.1
   - zarr >=2.3.2
   - sphinx >=2.0.1
   # Testing
   - flake8 >=3.7
   - pytest >=4.4
-  - pytest-cov >=2.6
-
+  - pytest-cov >=2.6
```
Comment (AliceBalfanz) [inline environment.yml]:
```suggestion
  - pytest-cov >=2.6
  
```

Comment (AliceBalfanz) [reply environment.yml]: New line missing at the end

**Pull request merged**